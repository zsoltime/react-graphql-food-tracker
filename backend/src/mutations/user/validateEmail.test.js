'use strict';

const validateEmail = require('./validateEmail');

describe('validateEmail mutation', () => {
  it('updates user email and returns data', async () => {
    const args = { token: 'testToken' };
    const fakeUser = {
      id: 'testId',
      email: 'test@email.io',
      pendingEmail: 'newTest@email.io',
      permissions: ['USER'],
      profile: null,
    };
    const mockedContext = {
      db: {
        mutation: {
          updateUser: jest.fn().mockImplementation(() => Promise.resolve({
            ...fakeUser,
            pendingEmail: null,
            email: fakeUser.pendingEmail,
            token: null,
            tokenExpiry: null,
          })),
        },
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([{ ...fakeUser }])),
        },
      },
    };

    const res = await validateEmail(null, args, mockedContext);

    expect(mockedContext.db.query.users).toHaveBeenCalledWith(
      expect.objectContaining({
        where: {
          token: args.token,
          tokenExpiry_gte: expect.any(Number),
        },
      })
    );
    expect(
      mockedContext.db.mutation.updateUser.mock.calls[0]
    ).toMatchSnapshot();
    expect(res).toMatchSnapshot();
  });

  it('throws error if token is invalid', () => {
    const args = { token: 'invalidToken' };
    const mockedContext = {
      db: {
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([])),
        },
      },
    };

    expect(
      validateEmail(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
