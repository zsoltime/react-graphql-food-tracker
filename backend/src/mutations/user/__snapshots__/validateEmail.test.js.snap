// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateEmail mutation throws error if token is invalid 1`] = `"Token is either invalid or expired"`;

exports[`validateEmail mutation updates user email and returns data 1`] = `
Array [
  Object {
    "data": Object {
      "email": "newTest@email.io",
      "pendingEmail": null,
      "token": null,
      "tokenExpiry": null,
    },
    "where": Object {
      "email": "test@email.io",
    },
  },
]
`;

exports[`validateEmail mutation updates user email and returns data 2`] = `
Object {
  "email": "newTest@email.io",
  "id": "testId",
  "pendingEmail": null,
  "permissions": Array [
    "USER",
  ],
  "profile": null,
  "token": null,
  "tokenExpiry": null,
}
`;
